DS-45 [가게] 엔티티 클래스 설계 및 정의

y00njinuk opened the pull request:
## Goals
- 가게 엔티티 클래스와 DTO 클래스를 정의한다.

## Implements
- [x] 가게 엔티티 클래스 정의
- [x] 가게 DTO 클래스 정의
- [x] 새로 정의된 스키마를 바탕으로 일부 누락된 필드 추가 및 업데이트 

## Related Issue 
- https://dangol-sonnim.atlassian.net/browse/DS-45

## Test
- "가게 DTO 클래스로부터 엔티티 클래스가 정상적으로 생성이 되야 한다."

y00njinuk committed:
feat: add categoryId to Store entity

[issue-key] DS-45
- Store 엔티티 클래스에 누락된 categoryId 필드 추가

y00njinuk committed:
feat: add updateName method

[issue-key] DS-45
- Store 엔티티 클래스에 가게이름을 변경하는 메서드 추가

y00njinuk committed:
feat: update annotation of StoreSignupRequestDTO

[issue-key] DS-45
- 생성자 관련된 어노테이션 추가
- 테스트 시에 해당 어노테이션을 통해 객체를 생성

y00njinuk committed:
feat: add storeRegisterNumber,storeRegisterName to Store entity

[issue-key] DS-45
- Store 엔티티 및 DTO 클래스에 사업자명,사업자번호 필드 추가

y00njinuk committed:
test: check to convert Store dto to entity class

[issue-key] DS-45
- 가게 DTO 클래스로부터 엔티티 클래스가 정상적으로 변환되었는지 확인
- 빌더 패턴을 활용하여 DTO 인스턴스 생성

y00njinuk commented on src/main/java/com/dangol/dangolsonnimbackend/store/domain/Store.java:
```diff
@@ -69,5 +69,6 @@ public Store(StoreSignupRequestDTO dto) {
         this.detailedAddress = dto.getDetailedAddress();
         this.comments = dto.getComments();
         this.officeHours = dto.getOfficeHours();
+        this.categoryId = dto.getCategoryId();
```
기존에 DTO에 정의된 필드 중 categoryId 필드가 누락이 되어 추가하였습니다.

y00njinuk commented on src/main/java/com/dangol/dangolsonnimbackend/store/domain/Store.java:
```diff
@@ -71,4 +71,8 @@ public Store(StoreSignupRequestDTO dto) {
         this.categoryId = dto.getCategoryId();
     }
+
+    public void updateName(String name) {
```
가게명을 변경하는 메소드를 새로 추가하였습니다.
이는 추후 진행된 Repo 테스트에서 활용이 될 예정입니다.

Hyeonziii replied:
update를 name만 진행하게 되는 건가요? 단순히 테스트에서 활용하기 위해 추가한 메소드인가요?

y00njinuk replied:
추후 서비스 단에서도 가게 이름을 수정하는 용도로도 활용될 예정입니다만 우선은 테스트 용도로 정의하였습니다.

y00njinuk commented on src/main/java/com/dangol/dangolsonnimbackend/store/dto/StoreSignupRequestDTO.java:
```diff
@@ -1,12 +1,14 @@
 @Setter
+@Builder
+@AllArgsConstructor
+@NoArgsConstructor
```
Getter/Setter 뿐만 아니라 생성자 및 빌더 어노테이션을 추가하였습니다.
DTO 인스턴스 생성 방식을 빌더 패턴으로 정의할 예정입니다.

y00njinuk commented on src/main/java/com/dangol/dangolsonnimbackend/store/domain/Store.java:
```diff
@@ -58,6 +58,12 @@ public class Store {
+    private String storeRegisterNumber;
+
+    @Column(nullable = false)
+    private String storeRegisterName;
```
새로 정의된 스키마를 바탕으로 사업자번호와 사업자명 필드를 추가하였습니다.

y00njinuk committed:
refactor: fix field name of Store entity

[issue-key] DS-45
- 수정된 의견을 바탕으로 필드 네이밍 수정
- Reference: https://github.com/jaemisam-a/dangol-sonnim-backend/pull/21#discussion_r1111128974

y00njinuk commented on src/main/java/com/dangol/dangolsonnimbackend/store/domain/Store.java:
```diff
@@ -24,7 +24,7 @@ public class Store {
 
     @Column(nullable = false)
-    private String storePhoneNumber;
+    private String phoneNumber;
```
https://github.com/jaemisam-a/dangol-sonnim-backend/pull/21#discussion_r1111128974 에서 진행된 내용을 바탕으로 필드 네이밍을 수정하였습니다.

rnjstmdals6 commented on src/main/java/com/dangol/dangolsonnimbackend/store/dto/StoreSignupRequestDTO.java:
```diff
@@ -34,4 +36,14 @@ public class StoreSignupRequestDTO {
+    private Long categoryId;
+
+    @NotNull(message = "사업자번호는 Null 일 수 없습니다.")
+    private String storeRegisterNumber;
```
Entity와 Dto의 필드 이름을 맞춰주면 좋을 것 같습니다

rnjstmdals6 commented on src/test/java/com/dangol/dangolsonnimbackend/store/domain/StoreTest.java:
```diff
@@ -11,26 +11,33 @@ public class StoreTest {
 
         Store store = new Store(dto);
 
         assertEquals(dto.getName(), store.getName());
```
하나만 비교하기로 했던 것 같습니다!

y00njinuk replied:
엇.. 다시 수정해놓겠습니다

y00njinuk committed:
refactor: fix filed name of Store entity

[issue-key] DS-45
- 가게 엔티티 클래스 필드 수정
